Add friends likes

DianaReshetnikova opened the pull request.

DianaReshetnikova committed:
часть функционала

DianaReshetnikova committed:
часть функционала-2

DianaReshetnikova committed:
Исправлены ошибки

DianaReshetnikova committed:
Исправлены ошибки-2

DianaReshetnikova committed:
Исправлены ошибки-3

DianaReshetnikova committed:
Исправлены ошибки-4

kriwsha commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -2,66 +2,61 @@
-@Slf4j
 public class FilmController {
-    private final Map<Long, Film> films = new HashMap<>();
+    private final FilmStorage filmStorage;
```
Контроллер не должен ничего знать о хранилище. Для этого используется слой сервисов. В этом суть слабой связки, лежащей в основе спринга. Представьте, что у вас поменяется хранилище, придется менять и сервис, и контроллер. А значит придется поддерживать две версии приложения. В случае слабой связки вы одним параметром будете указывать, какой сервис и хранилище использовать

kriwsha commented on src/main/java/ru/yandex/practicum/filmorate/controller/UserController.java:
```diff
@@ -1,67 +1,63 @@
+@RequiredArgsConstructor
 public class UserController {
-    private final Map<Long, User> users = new HashMap<>();
+    private final UserStorage userStorage;
```
И здесь от стораджа необходимо отказаться, а перенести в сервис

DianaReshetnikova committed:
Удалены объекты хранилищ фильмов и пользователей из контроллеров.
В сервисах FilmService и UserService созданы публичные методы доступа к методам их хранилищ
